hera_cal/io.py

```diff
@@ -7,26 +7,537 @@
+                    setattr(self, key, value)
+
+        elif self.filetype in ['miriad', 'uvfits']:
+            for meta in self.HERAData_metas:
+                setattr(self, meta, None)  # no pre-loading of metadata
+        else:
+            raise NotImplementedError('Filetype ' + self.filetype + ' has not been implemented.')
+
+    def reset(self):
+        '''Resets all standard UVData attributes, potentially freeing memory.'''
+        super(HERAData, self).__init__()
+
+    def get_metadata_dict(self):
+        ''' Produces a dictionary of the most useful metadata. Used as object
+        attributes and as metadata to store in DataContainers.
+
+        Returns:
+            metadata_dict: dictionary of all items in self.HERAData_metas
+        '''
+        antpos, ants = self.get_ENU_antpos()
+        ants = sorted(ants)
+        antpos = dict(zip(ants, antpos))
+
+        freqs = np.unique(self.freq_array)
+        times = np.unique(self.time_array)
+        lst_indices = np.unique(self.lst_array.ravel(), return_index=True)[1]
+        lsts = self.lst_array.ravel()[np.sort(lst_indices)]
+        pols = [polnum2str(polnum) for polnum in self.polarization_array]
+        antpairs = self.get_antpairs()
+        bls = [antpair + (pol,) for antpair in antpairs for pol in pols]
+
+        times_by_bl = {antpair: np.array(self.time_array[self._blt_slices[antpair]])
+                       for antpair in antpairs}
+        lsts_by_bl = {antpair: np.array(self.lst_array[self._blt_slices[antpair]])
+                      for antpair in antpairs}
+
+        locs = locals()
+        return {meta: eval(meta, {}, locs) for meta in self.HERAData_metas}
+
+    def _determine_blt_slicing(self):
+        '''Determine the mapping between antenna pairs and
+        slices of the blt axis of the data_array.'''
+        self._blt_slices = {}
+        for ant1, ant2 in self.get_antpairs():
+            indices = self.antpair2ind(ant1, ant2)
+            if len(indices) == 1:
+                self._blt_slices[(ant1, ant2)] = slice(indices[0], indices[0] + 1, self.Nblts)
+            elif not (len(set(np.ediff1d(indices))) == 1):
+                raise NotImplementedError('UVData objects with non-regular spacing of '
+                                          'baselines in its baseline-times are not supported.')
+            else:
+                self._blt_slices[(ant1, ant2)] = slice(indices[0], indices[-1] + 1,
+                                                       indices[1] - indices[0])
+
+    def _determine_pol_indexing(self):
+        '''Determine the mapping between polnums and indices
+        in the polarization axis of the data_array.'''
+        self._polnum_indices = {}
+        for i, polnum in enumerate(self.polarization_array):
+            self._polnum_indices[polnum] = i
+
+    def _get_slice(self, data_array, key):
+        '''Return a copy of the Nint by Nfreq waterfall or waterfalls for a given key. Abstracts
```

Comment:
It took me a little while after reading this to realize that `data_array` has the _full_ data size, and is not just the data_array specified by `key`. Perhaps add to the docstring that this function selects out the `key` part of `data_array`, just to be a little clearer..

Reply:
Done.